processor/eval/ledger_for_evaluator.go

```diff
@@ -139,6 +139,16 @@ func (l LedgerForEvaluator) GetAppCreator(indices map[basics.AppIndex]struct{})
 	return res, nil
 }
 
+// LookupKv is part of go-algorand's indexerLedgerForEval interface.
```

Comment:
This doesn't seem to be true. 
https://github.com/algorand/go-algorand/blob/master/ledger/evalindexer.go#L36

I guess we could add it, but I'm curious as to why you all have chosen not to include Boxes as [Creatables](https://github.com/algorand/go-algorand/blob/77cb33289d0cf02239db75c32cc2b3cfebf65354/data/basics/userBalance.go#L317)? Just from a naive perspective I would think Boxes should fit into that sort of model and so could follow similar lookup paths as Assets and Apps.

Reply:
You're correct that it's not true on master. However, I've added `LookupKv()` to the interface in the [companion PR](https://github.com/tzaffi/go-algorand/blob/zbox2/ledger/evalindexer.go#L46).

Boxes are associated to apps while Assets and Apps are associated with accounts. Since every app has its own account (basically, the hash of its index), one could in theory associate boxes as well to accounts directly. However, in the future, we may want to allow other sorts of key-value storage in the same `kvstore` ledger table so its more flexible to allow arbitrary keys for lookup, not just apps or accounts (e.g. maybe assets could have specialized boxes too in some future AVM). Creatables can never be re-created once they are deleted (yes, you can re-create an app with the same logic, but it has a new serial appid so it never truly re-creates the original app). At the end of the day, boxes can _outlive_ the app that they belonged to. This is a strange artifact that results in wasted space and min-balance allocation, but it is more evidence that boxes are fundamentally different from Creatables.

Reply:
Ok thanks for explaining. In terms of how Boxes are going to work this makes sense then. 

Conceptually it seems like the more logical way to register a Box would be to just associate it with an account and let application logic determine which of your account's Apps pull data from which of your account's Boxes, but that is clearly outside the scope of the changes here.

Reply:
As you suggest, it's logical to associate a box with an account. This reminds me of the issue I raised in the PR description regarding **Min Balance**. If we want to be able to support this feature for accounts and for app-accounts, and we want to avoid a future backfill, we may already in this PR or a follow-up before release, want to add an `app_account` column to `app_box` together with an index. @winder - appreciate your feedback as well.